### PR TITLE
Article: Laying out using Flexbox

### PR DESCRIPTION
**Pull request by CEREBR4L**
CSS-Flexbox Article

### CODE REVIEW

**Comment by Rafase282** [comment]
Aside form the commits, it looks good to me.

**Comment by Rafase282** [comment]
# :+1: LGTM

**Comment by CEREBR4L** [comment]
Seems it might even close: [#606](https://github.com/FreeCodeCamp/wiki/issues/606)

or at least make a start towards it (wasn't aware the issue was already created)

File: CSS-Flexbox.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+## CSS Flexbox 
+
+CSS Flexbox allows us to easily format our HTML like you never knew was possible, with flexbox it is possible to align vertically and horizontally, easily. Like the sound of it? Yeah me too.
```
Comment (alayek) [inline CSS-Flexbox.md]: > you never knew was possible, with flexbox it is

Why not break it over two separate sentences?

Comment (alayek) [inline CSS-Flexbox.md]: The first time you mention _CSS flexox_, link to official article on MDN.

File: CSS-Flexbox.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+[Flexyboxes](http://the-echoplex.net/flexyboxes/) is again a app that allows you too see code samples and change parameters to see how flexbox works visually.
+
+## Additional Resources
+[Mozilla Developer Network](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Flexible_Box_Layout/Using_CSS_flexible_boxes)
+
+[CSS Tricks](https://css-tricks.com/snippets/css/a-guide-to-flexbox/)
+
+
+Feel free to add to this document! Just look cheat sheets up and edit/complete this document!
```
Comment (alayek) [inline CSS-Flexbox.md]: Not needed

**Comment by alayek** [comment]
@abhisekp, you created #606, so you might wanna take a look at this :8ball: 

Barring some small issues, **LGTM** from my side.

**Comment by CEREBR4L** [comment]
New commit to address @alayek comments and also added video resources from #606

File: CSS-Flexbox.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
+## CSS Flexbox
```
Comment (abhisekp) [inline CSS-Flexbox.md]: Use Heading 1 or Heading 3

**Comment by abhisekp** [comment]
@CEREBR4L Please add some code examples along with some visual images for better understanding `flexbox` and also how, where and when to use them i.e. when is it appropriate to use `flexbox` and in what context.

File: CSS-Flexbox.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
+## CSS Flexbox 
+
+[CSS Flexbox](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Flexible_Box_Layout/Using_CSS_flexible_boxes) allows us to easily format our HTML like you never knew was possible. With flexbox it is possible to align vertically and horizontally, easily. Like the sound of it? Yeah me too.
+
+It is also fansastic for general layout of your website or app, it is easy to learn, supported well (in all modern broswers) and is great to work with, not to mention it doesn't take long to get to grips with at all!
```
Comment (theflametrooper) [inline CSS-Flexbox.md]: ~~broswers~~ browsers

Comment (robeerob) [inline CSS-Flexbox.md]: ~~fansastic~~ fantastic

File: CSS-Flexbox.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+## Games and Apps
+[Flexbox Defense](http://www.flexboxdefense.com/) is a web game that teaches flexbox the fun way.
+
+[Flexbox Froggy](http://flexboxfroggy.com/) is also a web game that teaches flexbox the fun way.
+
+[Flexbox in 5](http://flexboxin5.com/) is a web app that allows you to be able to see how flexbox works with a few simple controls.
+
+[Flexyboxes](http://the-echoplex.net/flexyboxes/) is again a app that allows you too see code samples and change parameters to see how flexbox works visually.
```
Comment (robeerob) [inline CSS-Flexbox.md]: ~~again a~~
~~a app~~ an app

**Comment by alayek** [comment]
@CEREBR4L please let us know if you need any help regarding this. Just fix the points raised by @abhisekp and @theflametrooper and we are good to go!

**Comment by CEREBR4L** [comment]
@alayek I have fixed formatting and highlighted spelling just... figuring out how I will write example and such.

**Comment by abhisekp** [comment]
@CEREBR4L another awesome site for flexbox http://www.flexboxpatterns.com

**Comment by abhisekp** [comment]
@CEREBR4L after completion, please comment here that you're complete so that the review process can continue. : )
Thanks for your contribution. :tada: :smiley: 

Let it keep coming. Love your work :heartpulse:

**Comment by theflametrooper** [comment]
## LGTM :+1:

**Comment by theflametrooper** [comment]
Thanks @CEREBR4L for that great article!

File: CSS-Flexbox.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+## Games and Apps
+[Flexbox Defense](http://www.flexboxdefense.com/) is a web game that teaches flexbox the fun way.
+
+[Flexbox Froggy](http://flexboxfroggy.com/) is also a web game that teaches flexbox the fun way.
+
+[Flexbox in 5](http://flexboxin5.com/) is a web app that allows you to be able to see how flexbox works with a few simple controls.
+
+[Flexyboxes](http://the-echoplex.net/flexyboxes/) is an app that allows you too see code samples and change parameters to see how flexbox works visually.
+
+[Flexbox Patters](http://www.flexboxpatterns.com) is a website that shows off loads of flexbox examples
```
Comment (abhisekp) [inline CSS-Flexbox.md]:
```diff
- [Flexbox Patters] ...
+ [Flexbox Patterns] ...
```

File: CSS-Flexbox.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+
+### Flex Flow
+
+Flex flow combines the use of `flex-wrap` and `flex-direction` into one property, it is used by first setting the direction and then the wrap. 
+
+`flex-flow: column wrap;` is an example if how to use this. 
+
+### Justify Content
+
+`justify-content` is a a property to align items in the container along the main axis (this changes depending on how content is displayed). There are multiple options for this and it allows us to fill any empty space on rows but defining how we want to 'justify' it.
```
Comment (abhisekp) [inline CSS-Flexbox.md]: ... is ~~a~~ a property ...

Comment (abhisekp) [inline CSS-Flexbox.md]:
```diff
- ... any empty space on ...
+ ... any empty spaces on ... 
```

File: CSS-Flexbox.md [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+
+![CSS playground no flex properties](http://i.imgur.com/m6rjD5n.png)
+
+After adding `display: flex;` we get the below, the default flex properties are applied causing it to show as such
+
+![CSS playground display flex default style](http://i.imgur.com/tjTiJtw.png)
+
+### Flex Direction
+
+`flex-direction:` allows us to control how the items in the container display, do you want them left to right, right to left, top to bottom or bottom to top? you can do all these easily with setting the flex direction of the container.
```
Comment (abhisekp) [inline CSS-Flexbox.md]:
```diff
+ ... to top? you can ...
- ... to top? You can ...
```

**Pull request merged**